server/src/main/java/org/opensearch/search/profile/aggregation/AggregationProfiler.java

```diff
@@ -47,29 +48,185 @@
  */
 public class AggregationProfiler extends AbstractProfiler<AggregationProfileBreakdown, Aggregator> {
 
-    private final Map<List<String>, AggregationProfileBreakdown> profileBreakdownLookup = new HashMap<>();
+    private final Map<Aggregator, AggregationProfileBreakdown> profileBreakdownLookup = new HashMap<>();
+    private final boolean concurrent;
```

Comment:
instead of having a boolean to decide on `concurrent` vs `non concurrent`  logic we can have a derived class named `ConcurrentAggregationProfiler` which can override `getTree` and have all the logic specific to concurrent path. Then the component constructing this class instance can create either `ConcurrentAggregationProfiler` or `AggregationProfiler` based on the use case.